zerver/lib/zcommand.py

```diff
@@ -73,4 +83,44 @@ def change_mode_setting(
                 setting_value=False,
             )
         )
+    elif command == "giphy":
+        if data is None:
+            raise JsonableError(_("Invalid data."))
+        api_key = settings.GIPHY_API_KEY
+        text = data.get("text")
+        stream_name = data.get("stream")
+        topic = data.get("topic")
+        recipient = data.get("recipient")
+
+        # Get allowed content rating
+        rating = "g"
+        options = user_profile.realm.GIPHY_RATING_OPTIONS
+        for option in options:
+            if options[option]["id"] == user_profile.realm.giphy_rating:
+                rating = option
+                break
+
+        if not api_key or rating == "disabled":
+            raise JsonableError(_("GIPHY integration is disabled. Please contact administrator to enable this feature."))
+
+        payload = {
+            "api_key": api_key,
+            "tag": str(text),  # Converted to str to fix mypy.
+            "rating": rating,
+        }
+        # Get a random GIF using GIPHY API related to the text.
+        response = requests.get("http://api.giphy.com/v1/gifs/random", params=payload)
+        if response.status_code == 200:
+            giphy_url = json.loads(response.text)["data"]["images"]["downsized_large"]["url"]
+            content = f"[{text}]({giphy_url})"
```

Comment:
@timabbott I do think we should send the text here since we are sending a random GIF related to the text.

Reply:
Do we need to implement /giphy on the backend?  I'd prefer for the slash command to do something frontend-only, since then we don't need to think about the security story for this logic.

Reply:
To provide support for other clients without them implementing the frontend logic themselves.